internal/cutil/cutil.go

```diff
@@ -0,0 +1,11 @@
+package cutil
+
+import "unsafe"
+
+// VoidPtr casts a uintptr value to an unsafe.Pointer value in order to use it
+// directly as a void* argument in a C function call.
+// CAUTION: NEVER store the result in a variable, or the Go GC could panic.
+func VoidPtr(i uintptr) unsafe.Pointer {
+	var nullPtr unsafe.Pointer
+	return unsafe.Pointer(uintptr(nullPtr) + i)
```

Comment:
What's this nullPtr do? Is it to avoid go vet errors?

Reply:
Yes, you can only cast a `uintptr` to `unsafe.Pointer` if you have cast it from `unsafe.Pointer` to `uintptr` in the same expression for pointer arithmetics.